server/src/pages/shotindex/view.js

```diff
@@ -293,6 +293,19 @@ class Card extends React.Component {
   constructor(props) {
     super(props)
     this.state = {panelOpen: "panel-closed", deleted: false};
+    this.maybeCloseDeleteConfirmation = this.maybeCloseDeleteConfirmation.bind(this);
```

Comment:
There is a lot of code duplication between these two view files. Could you extract the modal code to a single shared component?

Reply:
Yeah, that code duplication is pretty gross.  I'll try styling up a div as a button and have the confirmation dialog in there.  (Currently the dialog is actually a sibling of the button...which is the main reason they aren't in a component.)